tutorials/llm/llama-3/README.rst

```diff
@@ -0,0 +1,207 @@
+
+* `Authenticate with NVIDIA NGC <https://docs.nvidia.com/nim/large-language-models/latest/getting-started.html#ngc-authentication>`_, and download `NGC CLI Tool <https://docs.nvidia.com/nim/large-language-models/latest/getting-started.html#ngc-cli-tool>`_.
+
+
+`1. Creating a LoRA adapter with NeMo Framework <./llama3-lora-nemofw.ipynb>`__
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This notebook showcases performing LoRA PEFT on **Llama 3 8B Instruct** using a `PubMedQA <https://pubmedqa.github.io/>`__ using NeMo Framework. PubMedQA is a Question-Answering dataset for biomedical texts.
+
+To get started
+^^^^^^^^^^^^^^
+
+NeMo Framework is available as a `docker
+container <https://catalog.ngc.nvidia.com/orgs/nvidia/containers/nemo>`__.
+
+Run the container using the following command. It assumes that you have
+the notebook(s) available in the current working directory. If not,
+mount the appropriate folder to ``/workspace``.
+
+.. code:: bash
+
+   export FW_VERSION=24.05  # Make sure to choose the latest available tag
+
+
+.. code:: bash
+
+   docker run \
+     --gpus all \
+     --shm-size=2g \
+     --net=host \
+     --ulimit memlock=-1 \
+     --rm -it \
+     -v ${PWD}:/workspace \
+     -w /workspace \
+     -v ${PWD}/results:/results \
+     nvcr.io/nvidia/nemo:$FW_VERSION bash
+
+From within the container, start Jupyter lab:
+
+.. code:: bash
+
+   jupyter lab --ip 0.0.0.0 --port=8888 --allow-root
+
+Then, navigate to `this notebook <./llama3-lora-nemofw.ipynb>`__.
+
+`2. Multi-LoRA inference with NVIDIA NIM <./llama3-lora-deploy-nim.ipynb>`__
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This is a demonstration of deploying multiple LoRA adapters with NVIDIA
+NIM. NIM supports LoRA adapters in ``.nemo`` (from NeMo Framework), and
+Hugging Face model formats. We will deploy the PubMedQA LoRA adapter
+from the first notebook, alongside two other already trained LoRA adapters
+(`GSM8K <https://github.com/openai/grade-school-math>`__,
+`SQuAD <https://rajpurkar.github.io/SQuAD-explorer/>`__) that are
+available on NVIDIA NGC as examples.
+
+``NOTE``: While it’s not necessary to complete the LoRA training and
+obtain the adapter from the previous notebook (“Creating a LoRA adapter
+with NeMo Framework”) to follow along with this one, it is recommended
+if possible. You can still learn about LoRA deployment with NIM using
+the other adapters downloaded from NGC.
+
+
+1. Download example LoRA adapters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following steps assume that you have authenticated with NGC and downloaded the CLI tool, as mentioned in pre-requisites.
```

Comment:
The following steps assume that you have authenticated with NGC and downloaded the CLI tool, as listed in the Requirements section.